runtime/testing-hydra-dx/src/lib.rs

```diff
@@ -0,0 +1,1140 @@
+	type RegistrarOrigin = EnsureRootOrHalfCouncil;
+	type WeightInfo = ();
+}
+
+impl pallet_utility::Config for Runtime {
+	type Event = Event;
+	type Call = Call;
+	type WeightInfo = ();
+}
+
+/// ORML Configurations
+
+impl orml_tokens::Config for Runtime {
+	type Event = Event;
+	type Balance = Balance;
+	type Amount = Amount;
+	type CurrencyId = AssetId;
+	type WeightInfo = ();
+	type ExistentialDeposits = ExistentialDeposits;
+	type OnDust = ();
+	type MaxLocks = MaxLocks;
+}
+
+impl orml_currencies::Config for Runtime {
+	type Event = Event;
+	type MultiCurrency = Tokens;
+	type NativeCurrency = BasicCurrencyAdapter<Runtime, Balances, Amount, BlockNumber>;
+	type GetNativeCurrencyId = HDXAssetId;
+	type WeightInfo = ();
+}
+
+/// HydraDX Pallets configurations
+
+impl pallet_asset_registry::Config for Runtime {
+	type AssetId = AssetId;
+}
+
+impl pallet_xyk::Config for Runtime {
+	type Event = Event;
+	type AssetPairAccountId = pallet_xyk::AssetPairAccountId<Self>;
+	type Currency = Currencies;
+	type NativeAssetId = HDXAssetId;
+	type WeightInfo = pallet_xyk::weights::HydraWeight<Runtime>;
+	type GetExchangeFee = ExchangeFee;
+}
+
+impl pallet_claims::Config for Runtime {
+	type Event = Event;
+	type Currency = Balances;
+	type Prefix = ClaimMessagePrefix;
+	type WeightInfo = pallet_claims::weights::HydraWeight<Runtime>;
+	type CurrencyBalance = Balance;
+}
+
+impl pallet_exchange::Config for Runtime {
+	type Event = Event;
+	type AMMPool = XYK;
+	type Resolver = Exchange;
+	type Currency = Currencies;
+	type WeightInfo = pallet_exchange::weights::HydraWeight<Runtime>;
+}
+
+impl pallet_faucet::Config for Runtime {
+	type Event = Event;
+	type Currency = Currencies;
+}
+
+/// Staking pallets configurations
+pub mod impls;
+#[cfg(feature = "runtime-benchmarks")]
+use frame_benchmarking::frame_support::max_encoded_len::MaxEncodedLen;
+#[cfg(feature = "runtime-benchmarks")]
+use frame_benchmarking::frame_support::pallet_prelude::EnsureOrigin;
+#[cfg(feature = "runtime-benchmarks")]
+use frame_benchmarking::frame_support::pallet_prelude::{Get, Member};
+#[cfg(feature = "runtime-benchmarks")]
+use frame_benchmarking::frame_support::Parameter;
+use pallet_im_online::sr25519::AuthorityId as ImOnlineId;
+pub use pallet_staking::StakerStatus;
+use pallet_transaction_payment::TargetedFeeAdjustment;
+
+impl pallet_authorship::Config for Runtime {
+	type FindAuthor = pallet_session::FindAccountFromAuthorIndex<Self, Babe>;
+	type UncleGenerations = UncleGenerations;
+	type FilterUncle = ();
+	type EventHandler = (Staking, ImOnline);
+}
+
+pallet_staking_reward_curve::build! {
+	const REWARD_CURVE: PiecewiseLinear<'static> = curve!(
+		min_inflation: 0_040_000,
+		max_inflation: 0_080_000,
+		ideal_stake: 0_160_000,
+		falloff: 1_000_000,
+		max_piece_count: 40,
+		test_precision: 0_005_000,
+	);
+}
+
+parameter_types! {
+	pub const SessionsPerEra: sp_staking::SessionIndex = 6;
+	// Funds are bonded for 28 Days
```

Comment:
nope :P

Reply:
what values do you suggest?
```
SessionsPerEra = 24
BondingDuration = 2
SlashDeferDuration = 1          // SlashDeferDuration should be less than BondingDuration
```

Reply:
I meant the comment about 28 days. It's 60 minutes right now. Maybe era = session is not perfect also. We might want to do multiple sessions per era. Can you source some sane settings? Like 4 sessions = era.

Reply:
what about the following:

pub const EpochDuration: u64 = 10 * MINUTES as u64;
pub const SessionsPerEra: sp_staking::SessionIndex = 4;
// Funds are bonded for 32 hours
pub const BondingDuration: pallet_staking::EraIndex = 2;
pub const SlashDeferDuration: pallet_staking::EraIndex =  1;

Reply:
Sounds good

Reply:
done